src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/report/ReconciliationMapperTest.java

```diff
@@ -0,0 +1,154 @@
+        setupStorageConfig();
+        String content = Resources.toString(
+            getResource("reconciliation/valid-supplier-statement.json"),
+            UTF_8
+        );
+
+        LocalDate reportDate = LocalDate.now();
+        EnvelopeSupplierStatement envelopeSupplierStatement = new EnvelopeSupplierStatement(
+            UUID.randomUUID(),
+            reportDate,
+            content,
+            "1.0",
+            LocalDateTime.now()
+        );
+
+        ReconciliationStatement reconciliationStatement =
+            reconciliationMapper.convertToReconciliationStatement(
+                envelopeSupplierStatement,
+                "bulkscan"
+            );
+
+        assertThat(reconciliationStatement.date).isEqualTo(reportDate);
+        assertThat(reconciliationStatement.envelopes.size()).isEqualTo(3);
+        ReportedZipFile probateFile = reconciliationStatement.envelopes.get(0);
+        ReportedZipFile sscsFile = reconciliationStatement.envelopes.get(1);
+        ReportedZipFile cmcFile = reconciliationStatement.envelopes.get(2);
+        assertReportedZipFile(probateFile,
+            "1010404021234_14-08-2020-08-31.zip",
+            "probate",
+            null,
+            List.of("1015404021234", "1015404021235"),
+            List.of("123123", "123124")
+        );
+
+        assertReportedZipFile(sscsFile,
+            "9810404021234_14-08-2020-03-08-31.zip",
+            "sscs",
+            "121212_14-08-2020-03-08-21.zip",
+            List.of("9988774021234", "6655443301235"),
+            List.of("999999", "999234")
+        );
+
+        assertReportedZipFile(cmcFile,
+            "11111222333_14-05-2020-09-08-31.zip",
+            "cmc",
+            "9999999_4343_43343_4343.zip",
+            List.of("4444774020000", "5555443300001"),
+            List.of("3456789", "1234567")
+        );
+
+    }
+
+    private StorageConfigItem createStorageConfigItem(String targetContainer) {
+        StorageConfigItem storageConfigItem = new StorageConfigItem();
+        storageConfigItem.setTargetContainer(targetContainer);
+        return storageConfigItem;
+    }
+
+    private void assertReportedZipFile(
+        ReportedZipFile reportedZipFile,
+        String zipFileName,
+        String container,
+        String rescanFor,
+        List<String> scannableItemDcns,
+        List<String> paymentDcns
+    ) {
+        assertThat(reportedZipFile.zipFileName).isEqualTo(zipFileName);
+        assertThat(reportedZipFile.container).isEqualTo(container);
+        assertThat(reportedZipFile.rescanFor).isEqualTo(rescanFor);
+        assertThat(reportedZipFile.scannableItemDcns).isEqualTo(scannableItemDcns);
+        assertThat(reportedZipFile.paymentDcns).isEqualTo(paymentDcns);
+
+    }
+
+    private void setupStorageConfig() {
+        given(storageConfig.get("sscs")).willReturn(createStorageConfigItem("bulkscan"));
+        given(storageConfig.get("probate")).willReturn(createStorageConfigItem("bulkscan"));
```

Comment:
invalid config, target containers can't be the same.

Reply:
??

Reply:
target container names are same.

Reply:
containers are different, `target-storage-account` is the same

Reply:
what is wrong?